### ISSUE TITLE
utf8 handling segfault

### ISSUE THREAD

**Issue by ablakely** [eggheads/eggdrop#1326]
Hi,

I am running an eggdrop v1.9.3 with the bmotion script, whenever a user posts these characters in a channel the bot will segfault:
    
    𝙡𝙮
    
I haven't tested with others, I found this because a separate bot fetches the titles of urls for channel.

Log:

```
I am Kat, running eggdrop v1.9.3: 2 users (mem: 136k).
Online for 00:00 (status mode) - CPU: 00:00.75 - Cache hit:  0.0%
Configured with: 
Admin: ab3800
Config file: kat.conf
OS: Linux 5.10.0-10-amd64
Process ID: 1626260 (parent 1561580)
Tcl library: /usr/share/tcltk/tcl8.6
Tcl version: 8.6.11 (header version 8.6.11)
Tcl is threaded.
TLS support is enabled.
TLS library: OpenSSL 1.1.1k  25 Mar 2021
IPv6 support is enabled.
Threaded DNS core is enabled.
Socket table: 20/100
    Online as: Kat!~Kat@eph-ruk.vg3.0.127.IP (Kat)
    Connected to Ephasic [10.0.0.103]:6667 (connected for 00:00)
    Mode queue is at 0% (1 msgs)
    Server queue is at 1% (3 msgs)
    Active CAP negotiations: (none)
    #offtopic           :   5 members, enforcing "+tn" (greet)
    #lobby              : (not on channel)
Segmentation fault
```

**Comment by crazycatdevs** [comment]
This is a trouble with TCL which cannot treat utf8mb4.
The only actual way is to recompile TCL with the option TCL_UTF_MAX setted to 4 (or 6, but experimental).

**Comment by michaelortmann** [comment]
maybe we should add directions / instructions to an eggdrop doc? like doc/KNOWN-BUGS or doc/UTF8 if this is a useful fix/workaround?

can eggdrop detect how tcl was compiled regarding TCL_UTF_MAX during runtime and/or compile time?

**Comment by crazycatdevs** [comment]
Was discussed on [eggdrop.fr](https://forum.eggdrop.fr/Hack-anti-emoji-Unicode-t-1974.html) (sorry guys, french conversation), tcl way to detect is:
`set canUnicode [expr {[string length [encoding convertfrom utf-8 \xF0\x9F\xA4\x94]]==1}]`

**Comment by vanosg** [comment]
Fixed by #1402, now in 1.9.5
